packages/gafl-webapp-service/src/pages/multibuy/view-licences/__tests__/route.spec.js

```diff
@@ -94,19 +105,55 @@ describe('view licences > getData', () => {
     })
 
     const getSampleRequest = () => ({
-      ...mockRequest,
+      ...mockRequestPermission,
       i18n: {
         getCatalog: () => catalog
+      },
+      url: {
+        search: ''
+      }
+    })
+
+    const mockRequestNoPermission = createMockRequest({
+      cache: {
+        transaction: {
+          permissions: [emptyPermission]
+        }
+      }
+    })
```

Comment:
This, and above (mockRequestPermission, emptyPermission, permission) look like reused mocks, and you know my feelings on those ;)

It's all a bit confused - there's `mockRequestPermission`, but then `getSampleRequest` and `mockRequestNoPermission` as well as a new `getMockRequest` in addition to the imported `createMockRequest`. This is exactly what I'm talking about when I mention complicated code. It's not great in this file, but you've added significantly to the complexity buy just created new functions and file level variables that do very similar things to the existing mocks.

When you encounter a file like this, you need to cut through the mess. One function to create a mock request, one function to create a mock permission. _If_ (and only if) there's a good case for additional permission mocks, make sure you have a 'base' permission mock and then well named functions for your additional mocks, such as `getMockSeniorPermission`, `getMockJuniorPermission`, etc.

Don't use imported functions for creating mocks (i.e. `createMockRequest`), we could have just one imported function everywhere for creating a mock request, but then this would be incredibly complex and would hide what was actually going on. You should be able to see and understand your mocks, so you can build tests easily and quickly.